src/plugins/intel_gpu/src/graph/graph_optimizer/post_optimize_weights.cpp

```diff
@@ -185,97 +185,115 @@ void post_optimize_weights::add_gru_weights_reorder(primitive_id input_id, std::
 void post_optimize_weights::add_lstm_weights_reorder(primitive_id input_id, std::shared_ptr<WeightsReorderParams> reorder_params, program& p, \
                                                      cldnn::program_node& prev, cldnn::program_node& node, size_t i) {
     OPENVINO_ASSERT(reorder_params != nullptr, "[GPU] WeightsReorderParams is not initialized.");
-    std::string reorder_id = input_id + "_reo_" + std::to_string(i);
-    const auto dir_num = static_cast<int>(reorder_params->get_input_layout().get_shape()[0]);
-    auto hiddenSize = reorder_params->get_input_layout().get_shape()[1] / 4;
-    auto inputSize = static_cast<int>(reorder_params->get_input_layout().get_shape()[2]);
-    int size_third;
-    const int W_idx = 3;
-    if (i == W_idx) {
-        size_third = inputSize;
+
+    cache_key ckey{prev.id(), reorder_params->get_output_layout()};
+    auto itr = _cached_lstm_weights_reorder.find(ckey);
```

Comment:
could you reorganize the code for readability?
```
// If we already did the work, reuse existing node
if (itr != _cached_lstm_weights_reorder.end()) {
        node.replace_dependency(i, *itr->second, false);
        return;
}

// This is first time. Add nodes to reorder weights.
/* long code here */
```

Reply:
You're right. Updated it! Thanks!